hybrasyl/Throttle.cs

```diff
@@ -0,0 +1,283 @@
+﻿using Hybrasyl.Objects;
+using log4net;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Hybrasyl.Enums;
+
+namespace Hybrasyl
+{
+
+
+    public interface IThrottle
+    {
+        /// <summary>
```

Comment:
do we need to get all the properties at the interface level, or just at the class level?

Reply:
That's a good point, the properties can be moved to the `abstract`. Especially if we are making `IThrottle` / `IPacketThrottle`.

Reply:
+1